runtime/dart_vm.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_RUNTIME_DART_VM_H_
+#define FLUTTER_RUNTIME_DART_VM_H_
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "flutter/common/settings.h"
+#include "flutter/runtime/dart_isolate.h"
+#include "flutter/runtime/dart_snapshot.h"
+#include "flutter/runtime/service_protocol.h"
+#include "lib/fxl/build_config.h"
+#include "lib/fxl/functional/closure.h"
+#include "lib/fxl/macros.h"
+#include "lib/fxl/memory/ref_counted.h"
+#include "lib/fxl/memory/ref_ptr.h"
+#include "lib/fxl/memory/weak_ptr.h"
+#include "third_party/dart/runtime/include/dart_api.h"
+
+namespace blink {
+
+class DartVM : public fxl::RefCountedThreadSafe<DartVM> {
+ public:
+  class PlatformKernel;
+
+  FXL_WARN_UNUSED_RESULT
+  static fxl::RefPtr<DartVM> ForProcess(Settings settings);
+
+  static fxl::RefPtr<DartVM> ForProcessIfInitialized();
+
+  static bool IsRunningPrecompiledCode();
+
+  const Settings& GetSettings() const;
+
+  PlatformKernel* GetPlatformKernel() const;
+
+  const DartSnapshot& GetVMSnapshot() const;
+
+  const DartSnapshot& GetIsolateSnapshot() const;
```

Comment:
On Fuchsia, this can't be a singleton. Each AOT app has its own dylib with its own isolate snapshot.

Reply:
Converted this to a shared pointer and added a field for the same on `blink::DartIsolate`. However, there is still the issue of figuring out the snapshot for the service isolate. So the VM still has the default isolate snapshot buffer that is used for the service isolate, for all other isolates, the snapshot data is obtained from the blink::DartIsolate of the parent of the isolate being created.

Reply:
Updated tests.